File: javascript/libjoynr-js/src/main/js/joynr/exceptions/JoynrRuntimeException.js
/*
 * #%L
 * %%
 * Copyright (C) 2019 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
const JoynrException = require("./JoynrException");

class JoynrRuntimeException extends JoynrException {
    /**
     * Constructor of JoynrRuntimeException object used for reporting
     * error conditions. This serves as super class for other more specific
     * runtime exception objects and inherits from JoynrException.
     *
     * @param {Object} [settings] the settings object for the constructor call
     * @param {String} [settings.detailMessage] message containing details
     *            about the error
     * @returns {JoynrRuntimeException} The newly created IllegalAccessException object
     */
    constructor(settings = {}) {
        super(settings);

        /**
         * Used for serialization.
         * @name JoynrRuntimeException#_typeName
         * @type String
         */
        this._typeName = "joynr.exceptions.JoynrRuntimeException";
        this.name = "JoynrRuntimeException";
    }

    static _typeName = "joynr.exceptions.JoynrRuntimeException";
}

module.exports = JoynrRuntimeException;

File: java/javaapi/src/main/java/io/joynr/exceptions/JoynrRuntimeException.java
/*
 * #%L
 * %%
 * Copyright (C) 2011 - 2025 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package io.joynr.exceptions;

import jakarta.ejb.ApplicationException;

import com.fasterxml.jackson.annotation.JsonProperty;

@ApplicationException
public class JoynrRuntimeException extends RuntimeException implements JoynrException {
    private static final long serialVersionUID = 1L;

    public JoynrRuntimeException() {
        super();
    }

    public JoynrRuntimeException(String message) {
        super(message);
    }

    public JoynrRuntimeException(String message, Throwable cause) {
        super(message, cause);
    }

    public JoynrRuntimeException(Throwable cause) {
        super(cause);
    }

    @JsonProperty("detailMessage")
    @Override
    public String getMessage() {
        return super.getMessage();
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        JoynrRuntimeException other = (JoynrRuntimeException) obj;
        if (getMessage() == null) {
            if (other.getMessage() != null) {
                return false;
            }
        } else if (!getMessage().equals(other.getMessage())) {
            return false;
        }
        return true;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((getMessage() == null) ? 0 : getMessage().hashCode());
        return result;
    }
}
